filebeat/harvester/harvester.go

```diff
@@ -25,7 +26,8 @@ import (
 type Harvester struct {
 	Path               string /* the file path to harvest */
 	Config             *config.HarvesterConfig
-	Offset             int64
+	offset             int64
+	offsetLock         sync.Mutex
```

Comment:
why still required? Due to prospector asking for state in order to forward to registrar?